src/main/java/com/parasoft/report/transformer/XMLToSarif.java

```diff
@@ -164,4 +168,34 @@ private String[] avoidDuplicateProjectRootPaths(String[] paths) {
         }
         return uniquePaths.toArray(new String[0]);
     }
+
+    private void logUnconvertedPathsInSarifReport() {
+        if (this.projectRootPaths != null) {
+            ObjectMapper mapper = new ObjectMapper();
+            try {
+                JsonNode jsonNode = mapper.readTree(this.outputSarifReport);
+                JsonNode runsNode = jsonNode.get("runs");
+                Iterator<JsonNode> runsElements = runsNode.elements();
+                Set<String> uriSet = new HashSet<>();
+                runsElements.forEachRemaining(run -> {
+                    run.get("artifacts").elements().forEachRemaining(artifact -> {
+                        if (artifact.size() > 0) {
+                            JsonNode locationNode = artifact.get("location");
+                            String uri = locationNode.get("uri").asText();
+                            if (!locationNode.has("uriBaseId")) {
+                                uriSet.add(uri);
+                            }
+                        }
+                    });
+                });
+                List<String> uriList = new ArrayList<>(uriSet);
+                Collections.sort(uriList);
+                for (String uri : uriList) {
+                    Logger.info(MessageFormat.format("The unconverted paths in the SARIF report: {0}", uri));
```

Comment:
Instead of writing the same information message multiple times, let's write it once outside the for loop as:
`The following paths have not been converted to relative paths:`
Then, print each file path inside the for loop.